core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -50,6 +50,16 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   KOKKOS_IMPL_HIP_SAFE_CALL(
       hipGetDeviceProperties(&Impl::HIPInternal::m_deviceProp, hip_device_id));
   KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(hip_device_id));
+
+  // Check that we are running on the expected architecture
+  std::string arch_name = Impl::HIPInternal::m_deviceProp.gcnArchName;
+  if (arch_name != KOKKOS_ARCH_AMD_GPU) {
```

Comment:
```suggestion
  if (std::string arch_name = Impl::HIPInternal::m_deviceProp.gcnArchName;
      arch_name != KOKKOS_ARCH_AMD_GPU) {
```
(not formatted)
would prefer a string_view and a narrower scope for arch_name but not blocking

Reply:
string_view is annoying to use when creating the error message because you cannot concatenate it with `char`.